queries/subsquid/general/swapsList.graphql

```diff
@@ -0,0 +1,25 @@
+query swapsList($where: SwapWhereInput!, $orderBy: [SwapOrderByInput!] = [blockNumber_DESC]) {
+    swaps(where: $where, orderBy: $orderBy) {
+    id
+    expiration
+    blockNumber
+    status
+    caller
+    surcharge
+    price
+    nft {
+      id
+      name
+      sn
+      currentOwner
+      image
+    }
+    desired {
+      id
+      name
+      sn
+      currentOwner
+      image
+    }
+  }
```

Comment:
The difference between this graphql and `queries/subsquid/general/offersList.graphql` is only the field `surcharge`. Is it possible to create a fragment to reuse a bit?

Reply:
I had a look at it and from what I could find GraphQL doesn’t allow you to share a single fragment between different types since`Offer` and `Swap` don’t share a common interface or union.

cant't do something like 

````
fragment tradeDetails on Trade
````

or 
````
fragment tradeDetails on Offer, Swap
````

Reply:
I would propose unification/optimalization in next iteration.